### DISCUSSION TITLE
Error "ld: library 'clang_rt.osx' not found" after upgrading to Xcode 15

### DISCUSSION THREAD

**Discussion by tleyden** [tauri-apps/tauri#9468] [Q&A] [answered]
After upgrading to XCode 15, when I run `yarn tauri dev` I am getting the following build failure:

```
  = note: ld: warning: ignoring duplicate libraries: '-lSystem', '-lobjc'
          ld: warning: search path '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.3/lib/darwin' not found
          ld: library 'clang_rt.osx' not found
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

it is looking for a version of clang that is not present, since I now only have `15.0.0`:

```
ls -alh /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang
total 0
drwxr-xr-x   3 root  wheel    96B Feb  2 19:45 .
drwxr-xr-x  21 root  wheel   672B Feb 23 03:06 ..
drwxr-xr-x   5 root  wheel   160B Feb 23 03:03 15.0.0
```

How does Tauri determine which directories to put in the linker path?

**Comment by tleyden** [comment, marked as answer]
Cleaning the target directory fixed it:
```
cd src-tauri
rm -rf target
```

I also ran a `yarn upgrade` before that to upgrade `@tauri-apps/cli-darwin-arm64@1.5.9` to `@tauri-apps/cli-darwin-arm64@1.5.11`, but I don't think it made a difference.

Is manually deleting the `target` directory the best way to clean?  I didn't see a cli option for this.

**Reply by FabianLars** [reply]
Yeah it is. You can also run `cargo clean` in the src-tauri folder, but it's virtually the same as deleting it yourself.

**Reply by tleyden** [reply]
WDYT about adding `yarn tauri clean` to provide a consistent UX for `yarn` users?  That was the first place I looked for it.

Let me know if you'd take a PR.
